blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/Locator.java

```diff
@@ -85,6 +85,9 @@ public boolean equals(Locator other) {
     }
 
     public static Locator createLocatorFromPathComponents(String tenantId, String... parts) throws IllegalArgumentException {
+        if(StringUtils.isEmpty(tenantId) || parts == null)
+            throw new IllegalArgumentException("'tenantId' is null or empty OR 'parts' is null.");
```

Comment:
why are we throwing an exception here instead of logging an error?  
what will happen to the thread if an exception is thrown?  will the server go down?
i'm fine with checking inputs when received from the outside parties, but adding exceptions to this code seems dangerous without knowing the consequences.  am i missing something?

Reply:
Based on our discussion, I have replaced exception with log.error() call.